probe-rs-tools/src/bin/probe-rs/cmd/gdb_server/mod.rs

```diff
@@ -83,7 +84,27 @@ impl Cmd {
             }
             cmd.args(self.gdb_args);
             eprintln!("Spawning {cmd:?}");
-            Some(cmd.spawn()?)
+            let gdb = Arc::new(Mutex::new(cmd.spawn()?));
+
+            let gdb2 = gdb.clone();
+            tokio::spawn(async move {
+                let mut last_ctrl_c = Instant::now() - Duration::new(100, 0);
+                loop {
+                    // Don't exit on ctrl-c as you need to use this key combination
+                    // to ask gdb to interrupt execution of the tracee.
+                    tokio::signal::ctrl_c().await.unwrap();
+                    if last_ctrl_c.elapsed() < Duration::new(0, 500_000_000) {
+                        // Kill gdb if using ctrl-c twice within half a second.
+                        gdb2.lock().unwrap().kill().unwrap();
+                        println!();
+                        // Immediately exit to suppress error about gdb getting killer.
+                        std::process::exit(0);
```

Comment:
I'm not entirely sure why we have to terminate here, we are really impolite to the debugged device with this.

Reply:
Otherwise `bail!("Gdb failed with {exit_status}");` gets hit due to gdb getting killed by a signal. This results in an error with a length backtrace when `RUST_BACKTRACE=1` is used (which I have set in my shell profile).

Reply:
And is this the best we can do to quit gdb?

Reply:
Pushed a commit to instead suppress reporting errors returned by `gdb_server::run` after killing gdb.

Reply:
> And is this the best we can do to quit gdb?

I'm not familiar with the gdb remote protocol, but I think so. Closing the connection is not enough to make gdb stop. In fact gdb won't notice that the connection was closed until you do something that causes it to send a command to the gdbserver and once it does it just throws an error without exiting. Even if the command you used was `quit`, the error takes priority and you have to run `quit` a second time before gdb actually exits.

Reply:
I'm fine with ignoring ctrl-c, but what happens now if the user quits gdb normally? I'm not sure we should still keep ignoring ctrl-c in case the process has already exited. In fact, we should try and stop ourselves in that case, I think, if that's viable.

Reply:
If it was responsible for launching gdb, `probe-rs gdb` will already quit once the gdb it spawned. That is what the `gdb_process.try_wait()` in probe-rs-tools/src/bin/probe-rs/cmd/gdb_server/stub.rs is for.

Reply:
Makes sense, although in that case why do we get the signal at all? Anyway, ignoring that seems fair to me, thanks!